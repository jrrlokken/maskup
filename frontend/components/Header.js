import Link from 'next/link';
import styled from 'styled-components';

import Cart from './Cart';
import Nav from './Nav';
import Search from './Search';

const Logo = styled.h1`
  font-size: 4rem;
  margin-left: 2rem;
  position: relative;
  z-index: 2;
  /* background-color: var(--blue); */
  a {
    color: white;
    text-decoration: none;
    text-transform: uppercase;
    padding: 0.5rem 1rem;
  }
  @media (max-width: 1300px) {
    margin: 2rem;
    text-align: center;
  }
`;

const StyledHeader = styled.header`
  position: sticky;
  top: 0;
  z-index: 10;
  background-color: var(--blue);
  .bar {
    border-bottom: 10px solid var(--black);
    display: grid;
    grid-template-columns: auto 1fr;
    justify-content: space-between;
    align-items: stretch;
    @media (max-width: 1300px) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }
  
  .sub-bar {
    display: grid;
    grid-template-columns: 1fr auto;
    border-bottom: 1px solid var(--black);
  }
`;

const Header = () => (
  <StyledHeader>
    <div className="bar">
      <Logo><Link href='/'><a id='logo'>Mask Up!</a></Link></Logo>
      <Nav />
    </div>
    <div className="sub-bar">
      <Search />
    </div>
    <Cart />
  </StyledHeader>
);

export default Header;