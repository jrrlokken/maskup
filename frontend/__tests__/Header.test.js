import { configure, shallow } from 'enzyme';
import Adapter from '@hteker/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';

import Header from '../components/Header';

configure({ adapter: new Adapter() });

describe('<Header/>', () => {
  it('renders the logo and matches snapshotu', () => {
    const wrapper = shallow(
      <Header />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    const logo = wrapper.find('#logo');
    expect(logo.text()).toBe('Mask Up!');
  });
});
