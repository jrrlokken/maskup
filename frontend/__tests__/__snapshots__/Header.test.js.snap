// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders the logo and matches snapshot 1`] = `
<Header__StyledHeader>
  <div
    className="bar"
  >
    <Header__Logo>
      <Link
        href="/"
      >
        <a
          id="logo"
        >
          Mask Up!
        </a>
      </Link>
    </Header__Logo>
    <Nav />
  </div>
  <div
    className="sub-bar"
  >
    <Search />
  </div>
  <Cart />
</Header__StyledHeader>
`;

exports[`<Header/> renders the logo and matches snapshotu 1`] = `
<Header__StyledHeader>
  <div
    className="bar"
  >
    <Header__Logo>
      <Link
        href="/"
      >
        <a
          id="logo"
        >
          Mask Up!
        </a>
      </Link>
    </Header__Logo>
    <Nav />
  </div>
  <div
    className="sub-bar"
  >
    <Search />
  </div>
  <Cart />
</Header__StyledHeader>
`;
